sopel/tools/time.py

```diff
@@ -200,38 +225,44 @@ def format_time(db=None, config=None, zone=None, nick=None, channel=None,
     If ``db`` is not given or is not set up, steps 1 and 2 are skipped. If
     ``config`` is not given, step 3 will be skipped.
     """
-    utc = pytz.timezone('UTC')
-    tformat = None
+    target_tz: datetime.tzinfo = pytz.utc
+    tformat: Optional[str] = None
+
+    # get an aware datetime
+    if not time:
+        time = pytz.utc.localize(datetime.datetime.utcnow())
+    elif not time.tzinfo:
+        time = pytz.utc.localize(time)
+
+    # get target timezone
+    if zone:
+        target_tz = pytz.timezone(zone)
+
+    # get format for nick or channel
     if db:
         if nick:
             tformat = db.get_nick_value(nick, 'time_format')
         if not tformat and channel:
             tformat = db.get_channel_value(channel, 'time_format')
+
+    # get format from configuration
     if not tformat and config and config.core.default_time_format:
         tformat = config.core.default_time_format
+
+    # or default to hard-coded format
     if not tformat:
         tformat = '%Y-%m-%d - %T %z'
 
-    if not time:
-        time = datetime.datetime.now(tz=utc)
-    elif not time.tzinfo:
-        time = utc.localize(time)
-
-    if not zone:
-        zone = utc
-    else:
-        zone = pytz.timezone(zone)
-
-    return time.astimezone(zone).strftime(tformat)
+    # format local time with format
+    return time.astimezone(target_tz).strftime(tformat)
 
 
-def seconds_to_split(seconds):
+def seconds_to_split(seconds: int) -> Tuple[int, int, int, int, int, int]:
```

Comment:
This seems like it should have been a NamedTuple, whoops.

Reply:
I can probably do that yeah.